packages/roosterjs-editor-core/lib/corePlugins/ImageSelection.ts

```diff
@@ -64,6 +64,17 @@ export default class ImageSelection implements EditorPlugin {
                         }
                     }
                     break;
+                case PluginEventType.MouseDown:
+                    const mouseTarget = event.rawEvent.target;
+                    const mouseSelection = this.editor.getSelectionRangeEx();
+                    if (
+                        !safeInstanceOf(mouseTarget, 'HTMLImageElement') &&
+                        mouseSelection &&
+                        mouseSelection.type === SelectionRangeTypes.ImageSelection
```

Comment:
You just need to check if the target element is same with current selected image:

if (mouseSelection.type != Image || mouseSelection.image != event.rawEvent.target) {
  select(null)